### PR TITLE
WASM runtime switch to import memory

### PR DESCRIPTION
**Pull request by bkchr**
Up to now runtimes have exported their memory. To unify it with
sandboxing, this pr switches runtimes to import memory as well.

From a functional perspective, exporting/importing memory makes no
difference to the runtime.

To provide backwards compatibility, WASM exported memory is still supported.

### CODE REVIEW

File: client/executor/runtime-test/src/lib.rs [hunk 1]
```diff
@@ -56,7 +61,7 @@ sp_core::wasm_export_functions! {
 
 	fn test_empty_return() {}
 
-	fn test_exhaust_heap() -> Vec<u8> { Vec::with_capacity(16777216) }
+	fn test_exhaust_heap() -> Vec<u8> { Vec::with_capacity(18 * 64 * 1024) }
```
Comment (pepyakin) [inline client/executor/runtime-test/src/lib.rs]: The new value is `1179648`. I wonder why would we need such change?

File: primitives/runtime-interface/test/src/lib.rs [hunk 2]
```diff
@@ -41,7 +41,7 @@ fn call_wasm_method<HF: HostFunctionsT>(method: &str) -> TestExternalities {
 		sc_executor::WasmExecutionMethod::Interpreted,
 		&mut ext_ext,
 		&WASM_BINARY[..],
-		8,
+		1024,
```
Comment (pepyakin) [inline primitives/runtime-interface/test/src/lib.rs]: I wonder why we need this change?

File: client/executor/wasmtime/src/runtime.rs [hunk 3]
```diff
@@ -32,42 +32,38 @@ use sc_executor_common::{
...
 impl WasmRuntime for WasmtimeRuntime {
 	fn update_heap_pages(&mut self, heap_pages: u64) -> bool {
-		match heap_pages_valid(heap_pages, self.max_heap_pages) {
-			Some(heap_pages) => {
-				self.heap_pages = heap_pages;
-				true
-			}
-			None => false,
-		}
+		self.heap_pages as u64 == heap_pages
```
Comment (pepyakin) [inline client/executor/wasmtime/src/runtime.rs]: I wish I landed the chance that removes `update_heap_pages` separately...

Comment (bkchr) [reply client/executor/wasmtime/src/runtime.rs]: Yeah, we can do this later :)

File: client/executor/runtime-test/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply client/executor/runtime-test/src/lib.rs]: Ahh, I can probably revert this. Do you remember this question I had about initial pages?

I first had an implementation that used memory pages == heap pages, but now it is memory pages == intial + heap pages.

Tdlr, I revert this.

File: primitives/runtime-interface/test/src/lib.rs (hunk 2 above)
Comment (bkchr) [reply primitives/runtime-interface/test/src/lib.rs]: Some misunderstanding from me, I revert this as well.

File: client/executor/wasmtime/src/runtime.rs (hunk 3 above)
Comment (pepyakin) [reply client/executor/wasmtime/src/runtime.rs]: Yep, sure, I still hope my PR lands eventually lol

**Pull request merged**